### PR TITLE
✨ Add missing fields in ControlPlaneClass and MachineDeploymentClass

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PRs add missing fields in ControlPlaneClass and MachineDeploymentClass

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/7519
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/7520

### CODE REVIEW

File: api/v1beta1/clusterclass_types.go [hunk 1]
```diff
@@ -100,6 +100,26 @@ type ControlPlaneClass struct {
 	// referenced above is Machine based and supports setting replicas.
 	// +optional
 	MachineHealthCheck *MachineHealthCheckClass `json:"machineHealthCheck,omitempty"`
+
+	// NodeDrainTimeout is the total amount of time that the controller will spend on draining a node.
+	// The default value is 0, meaning that the node can be drained without any time limitations.
+	// NOTE: NodeDrainTimeout is different from `kubectl drain --timeout`
+	// NOTE: This value can be overridden while defining a Cluster.Topology using this ControlPlaneClass.
```
Comment (sbueringer) [inline api/v1beta1/clusterclass_types.go:107]:
```suggestion
	// NOTE: This value can be overridden while defining a Cluster.Topology.
```
nit: I would drop this part as users can't pick the "ControlPlaneClass"

(same below)

File: api/v1beta1/clusterclass_types.go [hunk 2]
```diff
@@ -125,6 +145,44 @@ type MachineDeploymentClass struct {
...
+	// NodeDrainTimeout is the total amount of time that the controller will spend on draining a node.
+	// The default value is 0, meaning that the node can be drained without any time limitations.
+	// NOTE: NodeDrainTimeout is different from `kubectl drain --timeout`
+	// NOTE: This value can be overridden while defining a Cluster.Topology using this MachineDeploymentClass.
+	// +optional
+	NodeDrainTimeout *metav1.Duration `json:"nodeDrainTimeout,omitempty"`
+
+	// NodeVolumeDetachTimeout is the total amount of time that the controller will spend on waiting for all volumes
+	// to be detached. The default value is 0, meaning that the volumes can be detached without any time limitations.
+	// NOTE: This value can be overridden while defining a Cluster.Topology using this CC.
```
Comment (sbueringer) [inline api/v1beta1/clusterclass_types.go:164]:
```suggestion
	// NOTE: This value can be overridden while defining a Cluster.Topology using this MachineDeploymentClass.
```

**Review by killianmuldoon — COMMENTED** [review]
Thanks for picking this up!

/lgtm

**Review by vincepri — APPROVED** [review]
/approve

**Pull request merged**